lib/js/src/manager/screen/utils/_PresentAlertOperation.js

```diff
@@ -0,0 +1,517 @@
+            this.onFinished();
+            return;
+        }
+        console.log('The presented alert was canceled successfully');
+        this.onFinished();
+    }
+
+    /**
+     * Creates an Alert from the _AlertView.
+     * @returns {Alert} - The Alert created from the AlertView.
+     */
+    createAlert () {
+        let alert = new Alert();
+        alert = this.assembleAlertText(alert);
+        alert.setDuration(this._alertView.getTimeout() * 1000);
+        if (this._alertView.getIcon() !== null && this._alertView.getIcon() !== undefined && this.supportsAlertIcon()) {
+            alert.setAlertIcon(this._alertView.getIcon().getImageRPC());
+        }
+        alert.setProgressIndicator(this._alertView.isShowWaitIndicator());
+        alert.setCancelID(this._cancelId);
+        if (this._alertView.getSoftButtons() !== null) {
+            const softButtons = [];
+            for (const button of this._alertView.getSoftButtons()) {
+                softButtons.push(button.getCurrentStateSoftButton());
+            }
+            alert.setSoftButtons(softButtons);
+        }
+
+        if (this._alertView.getAudio() !== null) {
+            const alertAudioData = this._alertView.getAudio();
+            alert.setPlayTone(alertAudioData.isPlayTone());
+            const ttsChunks = this.getTTSChunksForAlert(alertAudioData);
+
+            if (ttsChunks.length > 0) {
+                alert.setTtsChunks(ttsChunks);
+            }
+        }
+        return alert;
+    }
+
+    /**
+     * Pulls the TTSChunks from the AlertAudioData.
+     * @param {AlertAudioData} alertAudioData - An AlertAudioData instance.
+     * @returns {TTSChunk[]} - An array of TTSChunks.
+     */
+    getTTSChunksForAlert (alertAudioData) {
+        const ttsChunks = [];
+
+        if (this.supportsAlertAudioFile() && alertAudioData.getAudioFiles() !== null && alertAudioData.getAudioFiles() !== undefined && alertAudioData.getAudioFiles().length > 0) {
+            for (let index = 0; index < alertAudioData.getAudioFiles().length; index++) {
+                ttsChunks.push(new TTSChunk(alertAudioData.getAudioFiles()[index].getName(), SpeechCapabilities.FILE));
+            }
+        }
+
+        if (alertAudioData.getPrompts() !== null && alertAudioData.getPrompts() !== undefined && alertAudioData.getPrompts().length > 0) {
+            ttsChunks.concat(alertAudioData.getPrompts());
+        }
+        return ttsChunks;
+    }
+
+    /**
+     * Assemlbles the text for the Alert.
+     * @param {Alert} alert - An instance of Alert.
+     * @returns {Alert} - An instance of Alert.
+     */
+    assembleAlertText (alert) {
+        const nonNullFields = this.findNonNullTextFields();
+        if (nonNullFields.length === 0) {
+            return alert;
+        }
+        const numberOfLines = this._defaultMainWindowCapability === null ? 3 : _ManagerUtility.getMaxNumberOfAlertFieldLines(this._defaultMainWindowCapability);
```

Comment:
I don't think the default capability check needs to be here? Just the call to getMaxNumberOfAlertFieldLines